hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -55,16 +69,82 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         return Response.newBuilder().transactionGetReceipt(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var recordCache = context.createStore(RecordCache.class);
+        final TransactionGetReceiptQuery transactionGetReceiptQuery = query.transactionGetReceiptOrThrow();
+        if (transactionGetReceiptQuery.hasTransactionID()) {
+            final var receipts =
+                    recordCache.getReceipt(transactionGetReceiptQuery.transactionIDOrElse(TransactionID.DEFAULT));
+            mustExist(receipts, INVALID_TRANSACTION_ID);
+        } else {
+            throw new PreCheckException(INVALID_TRANSACTION_ID);
```

Comment:
I would have probably structured the method a little differently.

```
    @Override
    public void validate(@NonNull final QueryContext context) throws PreCheckException {
        requireNonNull(context);
        final var op = context.query().transactionGetReceiptOrThrow();
       
        // The transaction ID must be specified
        if (!op.hasTransactionID()) throw new PreCheckException(INVALID_TRANSACTION_ID);

        // The receipt must exist for that transaction ID
        final var recordCache = context.createStore(RecordCache.class);
        final var receipt = recordCache.getReceipt(op.transactionIDOrThrow());
        mustExist(receipt, INVALID_TRANSACTION_ID);
    }
```

You don't have to structure it this way. What I like about it is that it is easier for my head to manage because things get harder for me to comprehend when we get into if/else branches. Also, I moved the "recordCache" piece down to right next to where it is going to be used. Research shows that the shorter lifespan of a variable, the fewer bugs.

BTW, are we supposed to return INVALID_TRANSACTION_ID if the receipt isn't found? Or do we return a response more along the lines of "receipt not found"?

Reply:
that wha is happening in mono in `checkValidity`

`    @Override
    public ResponseCodeEnum checkValidity(final Query query, final StateView view) {
        final boolean isOk =
                (!defaultTxnId.equals(query.getTransactionGetReceipt().getTransactionID()));

        return isOk ? OK : INVALID_TRANSACTION_ID;
    }`

Reply:
I like also the structure you proposed it is very clean.